From 1087ec5dec7b43ca4679ae3ce510b65a5ae829c0 Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Wed, 23 Apr 2025 17:55:37 +0800
Subject: [PATCH] fixup! mtk: hostapd: add support for EPCS

Deinit epcs if epcs is init to prevent hostapd crash.
---
 hostapd/main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/hostapd/main.c b/hostapd/main.c
index cfdfb44542..bbd41ca998 100644
--- a/hostapd/main.c
+++ b/hostapd/main.c
@@ -834,6 +834,9 @@ static void hostapd_epcs_free(struct hapd_interfaces *ifaces)
 {
 	struct epcs_entry *entry, *n;
 
+	if (!ifaces->epcs.list.prev || !ifaces->epcs.list.next)
+		return;
+
 	dl_list_for_each_safe(entry, n, &ifaces->epcs.list,
 			      struct epcs_entry, list) {
 		dl_list_del(&entry->list);
-- 
2.45.2

